addons/hdfs-snapshot-mirroring/src/main/java/org/apache/falcon/retention/snapshots/HdfsSnapshotEvictor.java

```diff
@@ -0,0 +1,228 @@
+        }
+
+        // evict on target
+        retPolicy = cmd.getOptionValue(HdfsSnapshotMirrorProperties.TARGET_SNAPSHOT_RETENTION_POLICY.getName());
+        ageLimit = cmd.getOptionValue(
+                HdfsSnapshotMirrorProperties.TARGET_SNAPSHOT_RETENTION_AGE_LIMIT.getName());
+        numSnapshots = Integer.parseInt(
+                cmd.getOptionValue(HdfsSnapshotMirrorProperties.TARGET_SNAPSHOT_RETENTION_NUMBER.getName()));
+        if (retPolicy.equalsIgnoreCase("delete")) {
+            evictSnapshots(targetFs, targetDir, ageLimit, numSnapshots);
+        } else {
+            LOG.warn("Unsupported target retention policy {}", retPolicy);
+            throw new FalconException("Unsupported target retention policy " + retPolicy);
+        }
+
+        LOG.info("Completed HDFS Snapshot Eviction.");
+        return 0;
+    }
+
+    protected void evictSnapshots(DistributedFileSystem fs, String dirName, String ageLimit,
+                                  int numSnapshots) throws FalconException {
+        try {
+            LOG.info("Started evicting snapshots on dir {}{} using policy {}, agelimit {}, numSnapshot {}",
+                    fs.getUri(), dirName, ageLimit, numSnapshots);
+
+            long evictionTime = System.currentTimeMillis() - EvictionHelper.evalExpressionToMilliSeconds(ageLimit);
+
+            String snapshotDir = dirName + Path.SEPARATOR + ".snapshot" + Path.SEPARATOR;
+            FileStatus[] snapshots = fs.listStatus(new Path(snapshotDir));
+            if (snapshots.length <= numSnapshots) {
+                // no eviction needed
+                return;
+            }
+
+            // Sort by last modified time, ascending order.
+            Arrays.sort(snapshots, new Comparator<FileStatus>() {
+                @Override
+                public int compare(FileStatus f1, FileStatus f2) {
+                    return Long.compare(f1.getModificationTime(), f2.getModificationTime());
+                }
+            });
+
+            for (int i = 0; i < (snapshots.length - numSnapshots); i++) {
+                // delete if older than ageLimit while retaining numSnapshots
+                if (snapshots[i].getModificationTime() < evictionTime) {
+                    fs.deleteSnapshot(new Path(dirName), snapshots[i].getPath().getName());
+                }
+            }
+
+        } catch (ELException ele) {
+            LOG.warn("Unable to parse retention age limit {} {}", ageLimit, ele.getMessage());
+            throw new FalconException("Unable to parse retention age limit " + ageLimit, ele);
+        } catch (IOException ioe) {
+            LOG.warn("Unable to evict snapshots from dir {} {}", dirName, ioe);
+            throw new FalconException("Unable to evict snapshots from dir " + dirName, ioe);
+        }
+
+    }
+
+    private CommandLine getCommand(String[] args) throws org.apache.commons.cli.ParseException {
```

Comment:
I think few options are same for eviction and replication. Can we move it to util to reuse code?